### ISSUE TITLE
Automatically reformat descriptions for properties with complex types

### ISSUE THREAD

**Issue by BenjaminPelletier** [Apicurio/apicurio-studio#767]
Currently, if I supply a description for a property with a complex type, Apicurio produces YAML like:
```
my_property:
    $ref: '#/components/schemas/MyPropertyType'
    description: Some details about this specific property rather than its type
```
This is unfortunately invalid per the OpenAPI specification as discussed [here](https://github.com/OAI/OpenAPI-Specification/issues/556), but there is a (messy & ugly, IMO) workaround to rewrite that seems to be endorsed by Swagger and other tools.  The workaround would look like:
```
my_property:
    description: Some details about this specific property rather than its type
    allOf:
        - $ref: '#/components/schemas/MyPropertyType'
```
While this would make the Apicurio UI somewhat less consistent (#345), at least it would generate correct documentation (in both Swagger and ReDoc) whereas Apicurio's current output does not generate correct documentation.

As a way to partially address #345 at the same time, there could be a special check to see if a type has exactly one allOf $ref clause but no base $ref, and treat the allOf $ref clause as the singular type in that case.

**Comment by BenjaminPelletier** [comment]
I just noticed that the allOf workaround causes types to no longer be displayed in parentheses in ReDoc -- e.g., if I have
```
FooOperationURL:
    description: Endpoint that provides the ability to perform a foo operation
    type: string
```
```
preview_url:
    description: Endpoint at which to preview data
    allOf:
        - $ref: '#/components/schemas/FooOperationURL
```
...then ReDoc says `preview_url` is a `string` and doesn't mention anything about it being a FooOperationURL.  The current version of Apicurio produces
```
preview_url:
    description: Endpoint at which to preview data
    $ref: '#/components/schemas/FooOperationURL
```
...then ReDoc says `preview_url` is a `string (FooOperationURL)`, but its description is "Endpoint that provides the ability to perform a foo operation".

This seems like a really unfortunate no-win situation because [$ref-replacement is a poor way to specify types](https://github.com/OAI/OpenAPI-Specification/issues/556) so unfortunately I don't have a good suggestion for Apicurio alone :(  I've [raised the issue](https://github.com/Rebilly/ReDoc/issues/915) over at ReDoc, but this really just seems like a problem with OpenAPI.

**Comment by EricWittmann** [comment]
I agree that it seems like a failure with the spec.  It doesn't seem like it will be addressed very soon, since that thread you linked started in 2016...

**Comment by BenjaminPelletier** [comment]
Additional note: using `anyOf` instead of `allOf` causes ReDoc to generate documentation that is closer to correct, so I'm using that workaround currently.  I think it would be nice for Apicurio to automatically render as originally suggested in this issue, but use `anyOf` instead of `allOf`.

**Comment by adel-chouchane** [comment]
@EricWittmann Any news on this ?
